File: 백준/Silver/1764. 듣보잡/듣보잡.java
import java.io.*;
import java.util.*;

public class Main {

    private List<String> check(Set<String> neverHeard, String[] neverSeen) {
        List<String> names = new ArrayList<>();
        for(String name : neverSeen) {
            if (neverHeard.contains(name)) {
                names.add(name);
            }
        }
        return names;
    }


    public static void main(String[] args) throws IOException {

        Main T = new Main();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();


        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        Set<String> neverHeard = new HashSet<>();
        String[] neverSeen = new String[M];

        // 듣도 못한
        for (int i = 0; i < N; i++) {
            neverHeard.add(br.readLine());
        }

        // 보도 못한
        for(int i  =0; i < M; i++) {
            neverSeen[i] = br.readLine();
        }


        // 결과
        List<String> result = T.check(neverHeard, neverSeen);
        result.sort(String::compareTo); // 정렬

        sb.append(result.size()).append("\n");
        for (String s : result) {
            sb.append(s).append("\n");
        }

        // 출력
        System.out.println(sb.toString());
    }
}

File: 백준/Silver/1764. 듣보잡/듣보잡.cc
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <map>

using namespace std;

void solution(const vector<string>& neverSeen, const vector<string>& neverHeard){
    
    map<string, bool> map;
    vector<string> result;
    
    for(int i = 0; i < neverSeen.size(); i++){
        map.insert(make_pair(neverSeen[i], true));
    }
    
    for(int i = 0; i < neverHeard.size(); i++){
        if(map[neverHeard[i]]){
            result.push_back(neverHeard[i]);
        }
    }
    
    sort(result.begin(), result.end());
    
    cout << result.size() << endl;
    for(int i = 0; i < result.size(); i++){
        cout << result[i] << endl;
    }
    
}

int main() {
    
    int n, m;
    vector<string> neverHeard;
    vector<string> neverSeen;
    
    cin >> n >> m;
    
    for(int i = 0; i < n; i++){
        string input;
        cin >> input;
        neverHeard.push_back(input);
    }
    
    for(int i = 0; i < n; i++){
        string input;
        cin >> input;
        neverSeen.push_back(input);
    }
    
    
    solution(neverHeard, neverSeen);
    
    
    
    return 0;
}
